minoc/src/test/java/org/opencadc/minoc/ArtifactActionTest.java

```diff
@@ -147,10 +154,16 @@ public void testParsePath() {
             assertCorrectPath("token/cadc:TEST/myartifact", "cadc:TEST/myartifact", "token");
             assertCorrectPath("cadc:TEST/myartifact", "cadc:TEST/myartifact", null);
             assertCorrectPath("token/cadc:TEST/myartifact", "cadc:TEST/myartifact", "token");
-            assertCorrectPath("mast:long/uri/with/segments/fits.fits", "mast:long/uri/with/segments/fits.fits", null);
+            assertCorrectPath("mast:long/uri/with/segments/something.fits", "mast:long/uri/with/segments/something.fits", null);
             assertCorrectPath("token/mast:long/uri/with/segments/fits.fits", "mast:long/uri/with/segments/fits.fits", "token");
             assertCorrectPath("token-with-dashes/cadc:TEST/myartifact", "cadc:TEST/myartifact", "token-with-dashes");
             
+            assertCorrectPath("cadc:vault/uuid:fo/something.fits", "cadc:vault/uuid", null, "something.fits");
+            assertCorrectPath("token/cadc:vault/uuid:fo/something.fits", "cadc:vault/uuid", "token", "something.fits");
+            
+            assertCorrectPath("cadc:vault/uuid:/something.fits", "cadc:vault/uuid:/something.fits", null, null);
+
```

Comment:
How does it handle the `cadc:vault/uuid:fo/` case?

Reply:
this just tests that extra colons do not cause parts of the path to be lost (are preserved in the artifact uri); it is easy to implement the extraction in ways that lose stuff (eg if you split on colon and then it isn't a `:fo/`) 

the previous test checks that `:fo/` is extracted and becomes the override filename

Reply:
Sorry for being too vague, what I've meant is the incorrect case in which the URL ends in `:fo/`.